pkg/server/api/harvester/handlers.go

```diff
@@ -0,0 +1,225 @@
+	"github.com/HewlettPackard/galadriel/pkg/server/datastore"
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+)
+
+func (s *echoServer) syncBundleHandler(ctx echo.Context) error {
+	logger := ctx.Logger()
+
+	// auth
+	token := common.Member{
+		ID:          uuid.New(),
+		TrustDomain: spiffeid.RequireTrustDomainFromString("spiffe://td2"),
+		// ...
+	}
+	logger.Info("Receiving sync request")
+
+	body, err := io.ReadAll(ctx.Request().Body)
+	if err != nil {
+		logger.Error("failed to read body: ", err)
+		return err
+	}
+	receivedHarvesterState := common.SyncBundleBody{}
+	err = json.Unmarshal(body, &receivedHarvesterState)
+	if err != nil {
+		logger.Error("failed to unmarshal state: ", err)
+		return err
+	}
+
+	// get relationships for that member
+	relationships, err := s.DataStore.GetRelationship(context.TODO(), token.TrustDomain.String())
+	if err != nil {
+		logger.Error("failed to fetch relationships: ", err)
+		return err
+	}
+
+	response := common.SyncBundleResponse{}
+
+	currentState, err := s.calculateBundleState(relationships, token.TrustDomain.String())
+	if err != nil {
+		logger.Error("failed calculating bundle state: ", err)
+		return err
+	}
+
+	response.State = currentState
+	response.Update = s.calculateBundleSync(receivedHarvesterState.FederatesWith, relationships, token.ID)
+
+	responseBytes, err := json.Marshal(response)
+	if err != nil {
+		logger.Error("failed to marshal response: ", err)
+		return err
+	}
+
+	_, err = ctx.Response().Write(responseBytes)
+	if err != nil {
+		logger.Error("failed to write response: ", err)
+		return err
+	}
+
+	return nil
+}
+
+func (s *echoServer) postBundleHandler(ctx echo.Context) error {
+	logger := ctx.Logger()
+	logger.Info("Receiving post bundle request")
+
+	// auth
+	token := common.Member{
+		TrustDomain: spiffeid.RequireTrustDomainFromString("spiffe://td2"),
+	}
+
+	body, err := io.ReadAll(ctx.Request().Body)
+	if err != nil {
+		logger.Error("failed to read body: ", err)
+		return err
+	}
+
+	receivedHarvesterState := common.PostBundleBody{}
+	err = json.Unmarshal(body, &receivedHarvesterState)
+	if err != nil {
+		logger.Error("failed to unmarshal state: ", err)
+		return err
+	}
+
+	// fetch harvester state from db
+	currentState, err := s.DataStore.GetMember(context.TODO(), token.TrustDomain.String())
+	if err != nil {
+		logger.Error("failed to fetch member: ", err)
+		return err
+	}
+
+	// update internal state
+	if !bytes.Equal(receivedHarvesterState.TrustBundleHash, currentState.TrustBundleHash) {
+		_, err := s.DataStore.UpdateMember(context.TODO(), token.TrustDomain.String(), &common.Member{
+			TrustBundle:     receivedHarvesterState.TrustBundle,
+			TrustBundleHash: receivedHarvesterState.TrustBundleHash,
+		})
+		if err != nil {
+			logger.Error("failed to update member: ", err)
+			return err
+		}
+		logger.Infof("Trust domain %s has been successfully updated", receivedHarvesterState.TrustDomain)
```

Comment:
Same here. As a rule of thumb: use `Info` for high level component's lifecycle events and their relevant messages, and `Debug` for internal details that could be useful for someone looking deeper (bug hunting, performance issues, etc)
```suggestion
		logger.Debugf("Trust domain %s has been successfully updated", receivedHarvesterState.TrustDomain)
```